wurst/util/UnitUtils.wurst

```diff
@@ -0,0 +1,120 @@
+*/
+
+@configurable public constant real FIELD_OF_VIEW_FRONT  = 115.00  // Real FoV is: 180 - 2x (FIELD_OF_VIEW-90)	Using a value of 115 generates a FoV of 130   (natural FoV is about 135 degrees)
+@configurable public constant real FIELD_OF_VIEW_BEHIND = 115.00 
+
+/** Apply Buff to this unit for duration.*/
+public function unit.applyBuffTimed(buffTuple buffApplied, real duration)
+	this.addAbility(buffApplied.abilId)
+	this.setAbilityLevel(buffApplied.abilId, 1)
+	doAfter(duration) ->
+		if this != null
+			this.setAbilityLevel(buffApplied.abilId, 0)
+			this.removeAbility(buffApplied.abilId)
+			
+/** Increase Armor of a unit for a duration */
+public function unit.addArmorTimed(real armorAdd, real duration)
+	BlzSetUnitArmor(this, this.getArmor()+armorAdd)
+	doAfter(duration) ->
+		if this != null
+			BlzSetUnitArmor(this, this.getArmor()-armorAdd)
+			
+/** Increase Damage of a unit for a duration */		
+public function unit.addDamageTimed(int damageAdd, real duration)
+	BlzSetUnitBaseDamage(this, this.getBaseDamage(1)+damageAdd, 1)
+	doAfter(duration) ->
+		if this != null
+			BlzSetUnitBaseDamage(this, this.getBaseDamage(1)-damageAdd, 1)
+
+/** Make unit take DMG over time for given duration with hard-coded interval of 1s so dmg per seconds as input*/
+public function unit.applyDOTTimedFrom(unit damagingUnit, real duration, real dotDamage, buffTuple buffApplied)
+	this.applyDOTTimedFrom(damagingUnit, duration, 1, dotDamage, buffApplied)
+
+/** Make unit take DMG over time for given duration.*/
+public function unit.applyDOTTimedFrom(unit damagingUnit, real duration, real interval, real dotDamage, buffTuple buffApplied)
+	if this != null and this.isAlive()
+		this.addAbility(buffApplied.abilId)
+		this.setAbilityLevel(buffApplied.abilId, 1)
+
+		let targetUnit = this // saveback for closure
+		doPeriodicallyCounted(interval,  (duration/interval).floor()+1) cb ->
+			if targetUnit == null or not targetUnit.isAlive() or cb.isLast()
+				targetUnit.setAbilityLevel(buffApplied.abilId, 0)
+				targetUnit.removeAbility(buffApplied.abilId)
+				cb.stop()
+				destroy cb
+			else
+				if damagingUnit == null
+					targetUnit.damageTarget(targetUnit, dotDamage)
+				else
+					damagingUnit.damageTarget(targetUnit, dotDamage)
+					
+/** Make unit take mana and HP HEAL over time for given duration.*/
+public function unit.restoreHpMpTimed(real duration, real interval, real hotVal, real motVal, buffTuple buffApplied)
+	if this != null and this.isAlive()
+		this.addAbility(buffApplied.abilId)
+		this.setAbilityLevel(buffApplied.abilId, 1)
+
+		let targetUnit = this // saveback for closure
+		let maxCount   = (duration/interval).floor()+1	  //  last proc are is cut off 
+		doPeriodicallyCounted(interval, maxCount) cb ->	 // first proc will be applied after 1*interval
+			if targetUnit == null or not targetUnit.isAlive() or cb.isLast()
+				targetUnit.setAbilityLevel(buffApplied.abilId, 0)
+				targetUnit.removeAbility(buffApplied.abilId)
+				cb.stop()
```

Comment:
you don't need to stop before destroying I think

Reply:
True was only needed in a old version of stdlib, where it did not work properly..